tests/bears/DebugBearsTest.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import sys
+
+from io import StringIO
+
+from coalib.bears.Bear import Debugger, debug_run
+
+
+def func1(*args, **kwargs):
+    yield 1
+    yield 2
+    yield 3
+
+
+def func2(*args, **kwargs):
+    return([1, 2])
+
+
+def func3(*args, **kwargs):
+    kwargs['Makman'] = 'Admin'
+    return func1(*args, **kwargs)
+
+
+class DebugBearsTest(unittest.TestCase):
+    def setUp(self):
+        self.trace = sys.gettrace()
+        self.output = StringIO()
+
+    def tearDown(self):
+        sys.settrace(self.trace)
+
+    def test_debug_run_yield(self):
+        args = ('a', ('b', 'c'))
+        kwargs = {'d': 'e'}
+        input = StringIO('q\nc\nq\nc')
+        dbg = Debugger(stdin=input, stdout=self.output)
+        self.assertEqual(debug_run(func1, dbg, *args, **kwargs), [1, 2, 3])
+        output = self.output.getvalue()
```

Comment:
maybe create a helper function in the test that does exactly what you are doing here:
1. Create a debugger instance, with a list of commands to execute
2. run `debug_run`
3. return the text the debugger generated in the output

So it looks in the end like
```python
output = self.execute_debugger(['q', 'c', 'q', 'c'])
```

Reply:
Do I need to implement helper function now, can I do it after PR gets merge, I am already behind schedule
First phase work isn't completed yet

Reply:
Imo that's quite simple to do now:
```
def execute_debugger(debugger_commands, func, *args, **kwargs):
    input = StringIO('\n'.join(commands))
    output = StringIO()
    dbg = Debugger(stdin=input, stdout=output)
    debug_run(func, *args, **kwargs)
    return output.getvalue()
```
and then use it:
```
self.assertEqual(self.execute_debugger('q\nc\nq\nc', func1, *args, **kwargs))
```

Reply:
That's something you can reuse for all other test cases as well^^